### ISSUE TITLE
if cv_fun accepts argument x, this conflicts with future_lapply argument x

### ISSUE THREAD

**Issue by jeremyrcoyle** [tlverse/origami#24]
MRE:

```
do_nothing <- function(fold, x){
  #does nothing
}

folds <- make_folds(mtcars)
cross_validate(do_nothing, folds, x=mtcars)
```

Returns

> Error: is.function(FUN) is not TRUE

because `mtcars` is being passed as `x` to `future_lapply`, which is supposed to get folds

**Comment by nhejazi** [comment]
Easiest thing to do here is probably just write a check that gets called by `cross_validate` to stop when the cv_fun thrown in takes an argument `x`.

**Comment by HenrikBengtsson** [comment]
FYI and related to Issue #28; `future.apply::future_lapply()` uses `X` to match `base::lapply()` exactly;
```r
> args(future::future_lapply)
function (x, FUN, ..., future.globals = TRUE, future.packages = NULL, 
    future.seed = FALSE, future.lazy = FALSE, future.scheduling = 1) 
NULL

> args(future.apply::future_lapply)
function (X, FUN, ..., future.globals = TRUE, future.packages = NULL, 
    future.seed = FALSE, future.lazy = FALSE, future.scheduling = 1) 
NULL
```

This might make this issue moot; on the other hand another one may show up because of the name change.

**Comment by nhejazi** [comment]
Thanks for the update @HenrikBengtsson. `future.apply::future_lapply()` has been incorporated into `origami` as of 098a7e4 (currently a part of #27). After trying to confirm the bugfix, I'm still seeing this error, perhaps you could confirm this reproduces @jeremyrcoyle?

```
library(remotes)
do_nothing <- function(fold, X) {
  #does nothing
}

# test origami from current master (uses future::future_lapply)
remotes::install_github("jeremyrcoyle/origami@master")
folds <- origami::make_folds(mtcars)
origami::cross_validate(do_nothing, folds, X = mtcars)
## produces error: "Error: is.function(FUN) is not TRUE"

# test origami from PR for CRAN update (uses future.apply::future_lapply)
remotes::install_github("nhejazi/origami@fix-future_lapply")
folds <- origami::make_folds(mtcars)
origami::cross_validate(do_nothing, folds, X = mtcars)
## produces error: "Error: is.function(FUN) is not TRUE"
```

**Comment by jeremyrcoyle** [comment]
Fixed in #27 . Thanks @nhejazi . Will submit to CRAN today.
